capstone-backend/app.js

```diff
@@ -137,56 +138,340 @@ app.post ('/profile/interests', async (req, res) => {
+
+  await match.save()
+}
+
+function calculateScore(interest, category1, category2, weight1, weight2){
+  let user1Interest1 = [];
+  let user1Interest2 = [];
+  for (let i = 0; i < interest.user_1.length; i++){
+    user1Interest1 = user1Interest1.concat(interest.user_1[i].get(category1));
+    user1Interest2.push(interest.user_1[i].get(category2));
+  }
+  let user2Interest1 = [];
+  let user2Interest2 = [];
+  for (let i = 0; i < interest.user_2.length; i++){
+    user2Interest1 = user2Interest1.concat(interest.user_2[i].get(category1));
+    user2Interest2.push(interest.user_2[i].get(category2));
+  }
+  try {
+    let category1Score = compareRoles(user1Interest1, user2Interest1, weight1);
+    let category2Score = compareRoles(user1Interest2, user2Interest2, weight2);
+    let totalScore = category1Score + category2Score
+    return totalScore
+  }
+  catch (error){
+    console.log('error: ', error);
+    return 0;
+  }
+
+
+}
+
+function compareRoles(role1, role2, weight){
+  if (!role1 || !role2){
+    return 0;
+  }
+  let matches = 0;
+  for (let i = 0; i < role1.length; i++){
+    let match = false;
+    for (let j = 0; j < role2.length; j++){
+      if (role1[i] == role2[j]){
+        match = true;
+        break;
+      }
+    }
+    if (match){
+      matches++
+    }
+  }
+  let total = role1.length + role2.length;
+  if (total == 0){
+    return 0;
+  }
+  return ((matches/ total) * weight)
+}
+
+const skillWeight = 0.5;
+const skillNameWeight = 0.2;
+const rolesWeight = 0.3;
+
+function getScore(skills, roles){
+  const skillScore = calculateScore(skills, "category", "name", skillWeight, skillNameWeight);
+  const roleScore = compareRoles(roles.user_1, roles.user_2, rolesWeight);
+
+  return (skillScore + roleScore)
+}
+
+async function getMatches(currentUser){
+  const query = new Parse.Query(Parse.User);
+  query.notEqualTo("objectId", currentUser.id);
+  const entries = await query.find();
+
+  const Match = Parse.Object.extend("Match");
+  let count = 0;
+
+  entries.forEach(async entry => {
+    const matchInfo = await getUserData(entry);
+    const currentUserInfo = await getUserData(currentUser);
+    count++;
+    if ((!matchInfo.skills) || (!currentUserInfo.skills)){
+      return;
+    }
+
+    const skillsInfo = {user_1: currentUserInfo.skills, user_2: matchInfo.skills};
+    const rolesInfo = {user_1: currentUserInfo.roles, user_2: matchInfo.roles};
+    const matchScore = getScore(skillsInfo, rolesInfo);
+
+    const matchQuery = new Parse.Query(Match);
+    matchQuery.equalTo("user_1", currentUser.id);
+    matchQuery.equalTo("user_2", entry.id);
+    let matchResults = await matchQuery.first();
+
+    const matchQuery2 = new Parse.Query(Match);
+    matchQuery2.equalTo("user_2", currentUser.id);
+    matchQuery2.equalTo("user_1", entry.id);
+    let matchResults2 = await matchQuery2.first();
+
+    if (matchResults){
+      matchResults.set("score", matchScore);
+      matchResults2.set("score", matchScore);
```

Comment:
It doesn't look like you are doing anything with matchResults and matchResults2, they are not used bellow this line.
Perhaps I'm not understanding it correctly.